src/operator/src/statement/ddl.rs

```diff
@@ -643,42 +643,59 @@ impl StatementExecutor {
         drop_if_exists: bool,
         query_context: QueryContextRef,
     ) -> Result<Output> {
-        if let Some(table) = self
-            .catalog_manager
-            .table(
-                &table_name.catalog_name,
-                &table_name.schema_name,
-                &table_name.table_name,
-            )
+        // Reserved for grpc call
+        self.drop_tables(&[table_name], drop_if_exists, query_context)
             .await
-            .context(CatalogSnafu)?
-        {
-            let table_id = table.table_info().table_id();
-            self.drop_table_procedure(&table_name, table_id, drop_if_exists, query_context)
+    }
+
+    #[tracing::instrument(skip_all)]
+    pub async fn drop_tables(
+        &self,
+        table_names: &[TableName],
+        drop_if_exists: bool,
+        query_context: QueryContextRef,
+    ) -> Result<Output> {
+        let mut tables = Vec::with_capacity(table_names.len());
+        for table_name in table_names {
+            if let Some(table) = self
+                .catalog_manager
+                .table(
+                    &table_name.catalog_name,
+                    &table_name.schema_name,
+                    &table_name.table_name,
+                )
+                .await
+                .context(CatalogSnafu)?
+            {
+                tables.push(table.table_info().table_id());
+            } else if drop_if_exists {
+                // DROP TABLE IF EXISTS meets table not found - ignored
+                continue;
+            } else {
+                return TableNotFoundSnafu {
+                    table_name: table_name.to_string(),
+                }
+                .fail();
+            }
+        }
+
+        for (table_name, table_id) in table_names.into_iter().zip(tables.iter()) {
+            self.drop_table_procedure(table_name, *table_id, drop_if_exists, query_context.clone())
                 .await?;
 
             // Invalidates local cache ASAP.
             self.cache_invalidator
                 .invalidate(
                     &Context::default(),
                     &[
-                        CacheIdent::TableId(table_id),
+                        CacheIdent::TableId(*table_id),
                         CacheIdent::TableName(table_name.clone()),
```

Comment:
Can we avoid the `clone` here?

Reply:
For now, we can\`t move ownership from `&[TableName]`, maybe `Vec<TableName>` is ok.

In addition, the original implementation of `fn drop_table` here uses `table_name.clone()`:
https://github.com/GreptimeTeam/greptimedb/blob/a6269397c8924dbf9914ce13d81eded74a8c8f68/src/operator/src/statement/ddl.rs#L666
So I just follow it, for I`m not sure it is ok to remove it.